### ISSUE TITLE
Unhandled Exception when using 1-click Install

### ISSUE THREAD

**Issue by Omega207** [X-Hax/sa2-mod-loader#40]
I recently got the SA2 Mod Manager off of Gamebanana, but since using it I've had issues using the 1-click install option. Sometimes it does work, but otherwise it'll throw an error that seems to be related to timeouts with Gamebanana:

`Unhandled exception has occurred in your application...`

Exception Text
System.Net.WebException: The operation has timed out
   at System.Net.WebClient.DownloadDataInternal(Uri address, WebRequest& request)
   at System.Net.WebClient.DownloadString(Uri address)
   at System.Net.WebClient.DownloadString(String address)
   at ModManagerCommon.GameBananaItem.Load(String itemType, Int64 itemId) in C:\Programs\sa2-mod-loader\mod-loader-common\ModManagerCommon\GameBanana.cs:line 26
   at SA2ModManager.MainForm.HandleUri(String uri) in C:\Programs\sa2-mod-loader\SA2ModManager\MainForm.cs:line 292
   at SA2ModManager.MainForm.MainForm_Shown(Object sender, EventArgs e) in C:\Programs\sa2-mod-loader\SA2ModManager\MainForm.cs:line 390
   at System.Windows.Forms.Form.OnShown(EventArgs e)
   at System.Windows.Forms.Form.CallShownEvent()
   at System.Windows.Forms.Control.InvokeMarshaledCallbackDo(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbacks()
Loaded Assemblies
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4515.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
SA2ModManager
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///D:/Downloads/Sonic%20Adventure%202%20-%20Battle/SA2ModManager.exe
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4515.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4488.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4515.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4390.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
ModManagerCommon
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///D:/Downloads/Sonic%20Adventure%202%20-%20Battle/ModManagerCommon.DLL
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
Microsoft.GeneratedCode
    Assembly Version: 1.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
Newtonsoft.Json
    Assembly Version: 12.0.0.0
    Win32 Version: 12.0.2.23222
    CodeBase: file:///D:/Downloads/Sonic%20Adventure%202%20-%20Battle/Newtonsoft.Json.DLL

Installing mods manually works fine, though. It's just more of an annoyance having to manually import everything.
So, if anyone knows an answer to this issue, help would be much appreciated.
Thanks,
Omega207
P.S. Super unrelated, and I don't really care if nobody answers this question, but how does the level tester work? I've tried, but whenever I try to spawn in a location, it just puts me on the title screen. Is there something I'm missing?

**Comment by Omega207** [comment]
Seems to have eventually fixed itself.

**Comment by ItsEasyActually** [comment]
Meant to reply to this sooner. Still going to for the explanation. The error you experienced is with GameBanana's site. It's a rather common issue, usually happens on updates but can happen at any point if the GB servers are involved. 

Re: debug testing, you check the options you want applied and make your selections in the drop down menus then press the play button in that part of the window. Pressing save & play at the bottom of the modloader will boot the game normally.

**Comment by Omega207** [comment]
Thanks for replying; 
For GB, I figured it was kinda an issue with that. GB's servers are (sometimes notoriously) wonky, but even the next day, the servers  still had not (fully) fixed itself. That's why I didn't close this issue the day afterward.

As for the level tester, I tried it again, and it did work. I guess I _did_ accidentally select "Save & Play" instead of "Play" in the level selector. That was my bad... (But then again, I still didn't feel like it worked on the first two days...)
It's all fixed now, though.
